chainstate/client_test.go

```diff
@@ -41,6 +42,24 @@ func TestNewClient(t *testing.T) {
 		assert.Equal(t, customClient, c.HTTPClient())
 	})
 
+	t.Run("custom broadcast client", func(t *testing.T) {
+		arcConfig := broadcast_client.ArcClientConfig{
+			Token:  "",
+			APIUrl: "https://tapi.taal.com/arc",
+		}
+		customClient := broadcast_client.Builder().WithArc(arcConfig).Build()
+		require.NotNil(t, customClient)
+		c, err := NewClient(
+			context.Background(),
+			WithBroadcastClient(customClient),
+			WithMinercraft(&MinerCraftBase{}),
+		)
+		require.NoError(t, err)
+		require.NotNil(t, c)
+		assert.NotNil(t, c.BroadcastClient())
+		assert.Equal(t, customClient, c.BroadcastClient())
+	})
```

Comment:
It may look like we're just testing the getter, but I think in this case it's a bit more than that and it may be helpful to have those tests, because:

- it checks that the `WithBroadcastClient` method works correctly
- it checks if the custom client is properly initialised in `NewClient` method
- the application of custom client to new bux client is distributed across couple of files, so it's not just a simple function that you see in whole when you look at a file, but more like a small functionality
- if you look at the whole file `client_test.go`, there are all these "setter" options tested (such as `WithDebugging`, `WithHTTPClient`, etc) and I think we have them tested in other repos as well

WDYT?